statusHandler/presenter/chainInfoGetters.go

```diff
@@ -51,3 +55,89 @@ func (psh *PresenterStatusHandler) GetNumConnectedPeers() uint64 {
 func (psh *PresenterStatusHandler) GetCurrentRound() uint64 {
 	return psh.getFromCacheAsUint64(core.MetricCurrentRound)
 }
+
+// CalculateSynchronizationEstimation will calculate and return an estimation of
+// the time required for synchronization in a human friendly format
+func (psh *PresenterStatusHandler) CalculateSynchronizationEstimation() string {
+	currentBlock := psh.GetNonce()
+
+	psh.mutEstimationTime.Lock()
+	diffTime := time.Now().Sub(psh.startTime).Seconds()
+	blocksSynchronized := currentBlock - psh.startBlock
+	psh.mutEstimationTime.Unlock()
+
+	if blocksSynchronized == 0 {
+		return ""
+	}
+
+	probableHighestNonce := psh.GetProbableHighestNonce()
+	remainingBlocksToSynchronize := probableHighestNonce - currentBlock
+	timeEstimationSeconds := uint64(diffTime) * remainingBlocksToSynchronize / blocksSynchronized
+	remainingTime := secondsToHuman(int(timeEstimationSeconds))
+
+	return remainingTime
+}
+
+// CalculateSynchronizationSpeed will calculate and return speed of synchronization
+// how many block per second are synchronized
+func (psh *PresenterStatusHandler) CalculateSynchronizationSpeed() uint64 {
+	currentBlock := psh.GetNonce()
+	psh.mutEstimationTime.Lock()
+	blocksSynchronized := currentBlock - psh.startBlock
+	psh.mutEstimationTime.Unlock()
+	if blocksSynchronized == 0 {
```

Comment:
I think an estimation could be calculated even if no blocks were synchronized

Reply:
Another implementation for estimation was written.